jena-arq/src/main/java/org/apache/jena/query/Query.java

```diff
@@ -515,6 +519,14 @@ public void addHavingCondition(Expr expr)
         havingExprs.add(expr) ;
     }
 
+    // SELECT JSON
+
+    public Map<String, Object> jsonMapping = new LinkedHashMap<>();
```

Comment:
Preference - don't expose data structures which can be modified. Either return a copy (if it is not called repeatedly) or with a `Collections.unmodifiableMap` wrapper.

Reply:
I can't find any use of `jsonMapping` except serialization which would explain the lack of fields noted above.